src/sentry/api/serializers/snuba.py

```diff
@@ -337,3 +346,37 @@ def serialize(self, result, column="count", order=None):
             res["order"] = result.data["order"]
 
         return res
+
+    def merge_buckets(self, results):
+        """
+        Merges buckets together so that the total number of buckets we serialize doesn't
+        exceed `self.max_buckets`. For each merged bucket, we calculate the min/max/avg
+        for all values and return that as the result.
+        :param results:
+        :return:
+        """
+
+        # Determine the size of each bucket. In the case that results is not evenly
+        # divisible by `max_buckets` we'll just round up to the next whole number. This
+        # means that the last bucket will not have as many datapoints, but this
+        # shouldn't really change the result.
+        merged_bucket_size = int(ceil(len(results) / self.max_buckets))
```

Comment:
```suggestion
        merged_bucket_size = int(ceil(len(results) / float(self.max_buckets)))
```

I think you want this, otherwise Python will floor the division of the two ints.

Reply:
Also this returns warped results if `len(results) < 2 * self.max_buckets`. E.g. if I have 11 results and 10 max buckets, I'll only get 6 results back.

Reply:
>I think you want this, otherwise Python will floor the division of the two ints.

No need, this file does `from __future__ import division` which brings in py3 style division, so `(3 / 2) == 1.5`.

> Also this returns warped results if len(results) < 2 * self.max_buckets. E.g. if I have 11 results and 10 max buckets, I'll only get 6 results back.

I'm not sure what the best option is here. I'd prefer the results to be uniform, so that if we have any results that have been aggregated they all will be. I think that in practice this might be a huge problem, because we have a set of values we're using, but open to doing something better.